client/add/pokemon.ctrl.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Controller for handling the dialog to add a new box
+ * @return {function} A controller that contains 2 test elements
+ */
+module.exports = function ($mdDialog, Upload) {
+
+  this.visibilities = require('../../api/services/Constants.js').POKEMON_VISIBILITIES;
+  this.visibility = this.defaultPokemonVisibility;
+  this.box = this.defaultBox ? this.defaultBox.id : undefined;
+  this.file = {};
+
+  this.cancel = function() {
+    $mdDialog.cancel();
+  };
+
+  this.canAdd = function () {
+    return Boolean(this.visibility && this.file);
+  };
+
+  this.answer = function() {
+    this.file.upload = Upload.upload({
+      url: '/uploadpk6',
+      data: {visibility: this.visibility, pk6: this.file, box: this.box},
+    }).then(function (response) {
+      $mdDialog.hide(response.data);
+    }, function (response) {
+      console.log(response.status + ': ' + response.data);
+    }, function (evt) {
+      // Math.min is to fix IE which reports 200% sometimes
+      console.log(Math.min(100, parseInt(100.0 * evt.loaded / evt.total)));
```

Comment:
I like the idea of a progress meter, but these uploads are all around 250 bytes -- I'm not sure it's worth displaying it since it'll be at 100 almost immediately anyway.

Obviously, if we do decide to display it then we should probably do something other than printing it to the console.